### DISCUSSION TITLE
How to render legends

### DISCUSSION THREAD

**Discussion by jeremycohensolal-paylead** [avisi-cloud/structurizr-site-generatr#652] [Q&A] [answered]
Hello,
first of all congrats for this awesome tool!

My question is about generating a legend. How to achieve it ?

I use `"generatr.site.exporter" "structurizr"`
And in the rendered output there is no legend for different shapes.

How to enable it ?
It is supposed to be supported by almost all exporters https://arc.net/l/quote/yqsnjvvj

Regards

**Comment by jenspav** [comment, marked as answer]
Thanks for pointing this out. This is actually a difference between the C4-plantuml and the structurizr-plantuml exporter. The former renders the diagram as part of the diagram itself (as long as `c4plantuml.legend` is set to `true` which is the default), but the latter writes the legend separately which we currently ignore. 
If you want, you can create an issue for us to look at including the legend from structurizr-plantuml in the generated site. No promises though that it will be picked up. For the short term, you could switch to the c4-plantuml exporter.

**Reply by lucaspal** [reply]
I opened a PR that adds legends to diagrams, you can read more about it [here](https://github.com/avisi-cloud/structurizr-site-generatr/issues/654#issuecomment-4259016510)

**Comment by jeremycohensolal-paylead** [comment]
Thanks for your answer!
I will create an issue for that.

Is there a way to manually include the legend in a markdown file with a `!docs` ? that could be a totally acceptable workaround...

**Reply by jenspav** [reply]
No, we don't have something like this in place. If you add a markdown file to `!docs`, the first one appears in the info tab (https://avisi-cloud.github.io/structurizr-site-generatr/main/internet-banking-system/) and the others go to the documentation tab (https://avisi-cloud.github.io/structurizr-site-generatr/main/internet-banking-system/sections/).
